packages/quantic/force-app/main/default/lwc/quanticSearchBoxSuggestionsList/quanticSearchBoxSuggestionsList.js

```diff
@@ -87,18 +124,100 @@ export default class QuanticSearchBoxSuggestionsList extends LightningElement {
     }
   }
 
-  get suggestionsToRender() {
-    return this.suggestions?.map((s, i) => ({
-      ...s,
-      isSelected: this.selectionIndex === i,
-    }));
+  get allOptions() {
+    if (this.shouldDisplayRecentQueries) {
+      const options = [
+        ...this.filteredRecentQueries,
+        ...this.filteredSuggestions,
+      ]?.map((option, index) => ({
+        ...option,
+        key: index + 1,
+        isSelected: this.selectionIndex === index + 1,
+        containerCSSClass: `${optionCSSClass} ${this.selectionIndex === index + 1 ? 'slds-has-focus' : ''}`,
+        icon: option.isRecentQuery ? 'utility:clock' : 'utility:search',
+      }));
+      return [
+        {
+          key: 0,
+          isSelected: this.selectionIndex === 0,
+          isClearRecentQueryButton: true,
+        },
+        ...options,
+      ].slice(0, this.maxNumberOfSuggestions + 1);
```

Comment:
ok the index+ 1 here is because of the recent searches label + clear button option? 
nice!

Reply:
indeed.